lib/forkhelpers.ml

```diff
@@ -41,8 +41,8 @@ let waitpid (sock, pid) =
 	  | Fe.Finished (Fe.WSTOPPED n) -> (pid,Unix.WSTOPPED n)
 	end
 
-let waitpid_nohang ((sock, _) as x) =
-	(match Unix.select [sock] [] [] 0.0 with
+let waitpid_nohang ((sock, _) as x) timeout =
+	(match Unix.select [sock] [] [] timeout with
 	  | ([s],_,_) -> waitpid x
```

Comment:
The timeout applies to waiting for the file descriptor `sock` to become ready for reading, but then it looks to me as if the call to `waitpid` could take an arbitrarily long time, depending on how long the read takes in `Fecomms.read_raw_rpc`.
Therefore I don't think this achieves the intended behaviour of applying a timeout to the `waitpid` call as a whole.

The other thing I wonder about is what should be done to the spawned process if we reach the timeout. Should we try to kill it? Probably in most cases, but I don't think this library always has the information it needs to decide. Perhaps the caller who specifies the timeout should also supply an indication of what to do on reaching the timeout. The `finally` function could be useful here.